service/discovery/azure/compute.go

```diff
@@ -232,25 +232,25 @@ func (d *azureComputeDiscovery) handleFunction(function *armappservice.Site) voc
 		}
 
 		// Check all runtime versions to get the used runtime language and runtime version
-		if config.Properties.JavaVersion != nil {
+		if *config.Properties.JavaVersion != "" {
```

Comment:
Does that make sense?! Won't this crash if the property is not set? shouldn't we use our util.Deref here at least?

Reply:
Also, I wouldn't change too many things in this PR that have nothing to do with the original "problem".